### PR TITLE
Native Deepviews

### PR DESCRIPTION
**Pull request by natario1**
Changes the deepview implementation, now using native views instead of loading a web url inside `WebView`.

To test this:
- run the demo app
- long press on a link result to trigger its Deepview

### CODE REVIEW

File: BranchSearchSDK/src/main/java/io/branch/search/BranchDeepViewFragment.java [hunk 1]
```diff
@@ -83,64 +76,171 @@ public Dialog onCreateDialog(Bundle savedInstanceState) {
...
-        webSettings.setJavaScriptEnabled(false);
-
+        final BranchLinkResult link;
         Bundle args = getArguments();
-        Uri uri;
-        if (args != null && (uri = args.getParcelable(KEY_DESTINATION)) != null) {
-            mWebView.loadUrl(uri.toString());
-            mWebView.setWebViewClient(new WebViewClient() {
+        if (args == null || (link = args.getParcelable(KEY_LINK)) == null) {
+            throw new IllegalStateException("No link!");
```
Comment (bklastaitis-branch) [inline BranchSearchSDK/src/main/java/io/branch/search/BranchDeepViewFragment.java]: Do we really want to throw an exception? As an SDK, shouldn't we just fail silently?

File: BranchSearchSDK/src/main/java/io/branch/search/BranchDeepViewFragment.java [hunk 2]
```diff
@@ -83,64 +76,171 @@ public Dialog onCreateDialog(Bundle savedInstanceState) {
...
+        // Description
+        TextView description = view.findViewById(R.id.branch_deepview_description);
+        if (description != null) loadText(description, link.getDescription());
+
+        // Image
+        ImageView image = view.findViewById(R.id.branch_deepview_image);
+        if (image != null) {
+            String url = link.getImageUrl();
+            if (url != null
+                    && url.equals(link.getAppIconUrl())
```
Comment (bklastaitis-branch) [inline BranchSearchSDK/src/main/java/io/branch/search/BranchDeepViewFragment.java]: I'm not 100% sure what this is meant to do but the condition seems weird and makes me think it's leftover debug code. What is "=s90"? (Should probably be a variable, so it is clearer.) Also, "=s90" is only the value of the parameter, if we are removing it, should we remove the key as well?

File: BranchSearchSDK/src/main/java/io/branch/search/BranchDeepViewFragment.java [hunk 3]
```diff
@@ -83,64 +76,171 @@ public Dialog onCreateDialog(Bundle savedInstanceState) {
...
-                        return true;
-                    } else if (url.contains("play.google.com")) {
-                        Intent intent = new Intent(Intent.ACTION_VIEW, Uri.parse(url));
-                        startActivity(intent);
-                        return true;
-                    } else {
-                        return false;
-                    }
+                public void onClick(View v) {
+                    String url = "https://play.google.com/store/apps/details?id="
```
Comment (bklastaitis-branch) [inline BranchSearchSDK/src/main/java/io/branch/search/BranchDeepViewFragment.java]: "https://play.google.com/store/apps/details?id=" should be a variable

File: BranchSearchSDK/src/main/java/io/branch/search/BranchDeepViewFragment.java [hunk 4]
```diff
@@ -83,64 +76,171 @@ public Dialog onCreateDialog(Bundle savedInstanceState) {
...
+                            + link.getDestinationPackageName();
+                    Intent intent = new Intent(Intent.ACTION_VIEW, Uri.parse(url));
+                    startActivity(intent);
+                    dismiss();
                 }
             });
         } else {
-            throw new IllegalStateException("No destination!");
+            throw new IllegalStateException("Call to action button is missing!");
```
Comment (bklastaitis-branch) [inline BranchSearchSDK/src/main/java/io/branch/search/BranchDeepViewFragment.java]: Shouldn't we fail silently?

File: BranchSearchSDK/src/main/java/io/branch/search/BranchDeepViewFragment.java (hunk 2 above)
Comment (natario1) [reply BranchSearchSDK/src/main/java/io/branch/search/BranchDeepViewFragment.java]: I think @rkhaitan-branch can explain this better. But in short app icon urls come from a provider that adds `=s90` at the end of the URL and this makes them 90x90 (I think). Removing this for full-screen images makes them look much better. Rishi is very confident with using this in production. Will make this a variable 👍

File: BranchSearchSDK/src/main/java/io/branch/search/BranchDeepViewFragment.java (hunk 3 above)
Comment (natario1) [reply BranchSearchSDK/src/main/java/io/branch/search/BranchDeepViewFragment.java]: Will do

File: BranchSearchSDK/src/main/java/io/branch/search/BranchDeepViewFragment.java (hunk 4 above)
Comment (natario1) [reply BranchSearchSDK/src/main/java/io/branch/search/BranchDeepViewFragment.java]: This means that the developer has overridden our layout and passed in a layout that has no CTA button. This leaves with a dysfunctional UI (a deepview that goes nowhere) so I think we should instruct the developer do the right thing. What do you think @bklastaitis-branch ?

File: BranchSearchSDK/src/main/java/io/branch/search/BranchDeepViewFragment.java (hunk 1 above)
Comment (natario1) [reply BranchSearchSDK/src/main/java/io/branch/search/BranchDeepViewFragment.java]: I will remove this! It cannot happen anyway.

File: BranchSearchSDK/src/main/java/io/branch/search/BranchDeepViewFragment.java (hunk 4 above)
Comment (bklastaitis-branch) [reply BranchSearchSDK/src/main/java/io/branch/search/BranchDeepViewFragment.java]: I see, so the developer has the option to override, in that case yeah, informing them sounds like the right approach 👍

**Pull request merged**